src/VisualStudio/Core/Def/Implementation/InheritanceMargin/MarginGlyph/InheritanceMargin.xaml

```diff
@@ -215,7 +84,172 @@
     </Button.Template>
 
     <Button.ContextMenu>
-        <!-- Ensure the context menu is not null -->
-        <ContextMenu Style="{StaticResource ContextMenuStyle}"/>
+        <ContextMenu ItemsSource="{Binding MenuItemViewModels}"
+                     Style="{StaticResource ContextMenuStyle}"
+                     ItemContainerTemplateSelector="{StaticResource TemplateSelector}"
```

Comment:
This referencing the customized TemplateSelector,
so when it sees HeaderViewModel, it creates a header,
when it sees TargetViewModel, it creates a target,
when it sees MemberViewModel, it create a member